src/libraries/Common/src/Interop/Unix/System.Net.Security.Native/Interop.Initialization.cs

```diff
@@ -13,23 +13,42 @@ static Ssl()
         {
             SslInitializer.Initialize();
         }
+
+        // access for static properties that may be called before static constructor.
+        internal static void EsureInitialized()
+        {
+            SslInitializer.Initialize();
+        }
     }
 
     internal static class SslInitializer
     {
 #if !SYSNETSECURITY_NO_OPENSSL
+        private static readonly object _initLock = new object();
+        private static bool _initialized;
+
+#pragma warning disable CA1810
         static SslInitializer()
         {
-            CryptoInitializer.Initialize();
-
-            //Call ssl specific initializer
-            Ssl.EnsureLibSslInitialized();
-            if (Interop.Crypto.ErrPeekLastError() != 0)
+            lock (_initLock)
```

Comment:
None of this locking is necessary, or really even useful.   It can run in parallel across (today) 4 different assemblies.  Even after I'm done collapsing things it'll still be two.  So the called code is (supposed to be) defensive against concurrency.

Reply:
I know it _can_ run. But that will make it visible to `strace` or run with LD_DEBUG, would it not? 
What I'm trying to argue is that we can do unnecessary file system operation and system calls.

Reply:
The library load will only ever happen once, the P/Invoke into EnsureLibSslInitialized will only happen once per assembly (cctor guarantee), and the cascaded EnsureOpenSslInitialized will only happen once per assembly.  If you do the change as currently structured but remove the locking then there might be two calls to EnsureLibSslInitialized (one from the cctor, one from the extra explicit call).

Restructuring things to move the field off this class will keep things simple looking while staying with the current initialize-count.  Even just making a nested type will fix it... e.g. `Interop.Ssl.Tls13Supported` => `Interop.Ssl.Capabilities.Tls13Supported`.

Reply:
> Even just making a nested type will fix it... e.g. Interop.Ssl.Tls13Supported => Interop.Ssl.Capabilities.Tls13Supported

This makes sense to me.